opentracing-impl-java8/src/main/java/io/opentracing/propagation/SpanImpl.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.propagation;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import io.opentracing.Span;
+
+final class SpanImpl implements Span {
+
+    final Map<String,Object> traceState = new HashMap<>();
```

Comment:
I like a lot of this about this PR, but this is not one of them... I'd rather that we either (a) really commit to a "trace state" representation and make this a less-general PR that introduces `basictracer-java` (i.e., where traceState is explicitly `<trace_id, span_id, sampled>`), OR (b) we pull back the scope a bit and focus on things like the injector/extractor registry.